zookeeper-server/src/main/java/org/apache/zookeeper/server/SyncRequestProcessor.java

```diff
@@ -102,103 +108,95 @@ public void run() {
 
             // we do this in an attempt to ensure that not all of the servers
             // in the ensemble take a snapshot at the same time
-            int randRoll = r.nextInt(snapCount/2);
+            int randRoll = ThreadLocalRandom.current().nextInt(snapCount / 2, snapCount);
```

Comment:
I remember we changed to use Random from thread local once since it's approximately giving the same randomize after JDK 8 if I remember correctly.

Reply:
> j.u.Random continues to be an embarrassment, and replacing uses with ThreadLocalRandom wherever we can is progress.

So says the Java dev team.
https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8218282

> When applicable, use of ThreadLocalRandom rather than shared Random objects in concurrent programs will typically encounter much less overhead and contention.

>This class also provides additional commonly used bounded random generation methods.

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html